src/main/java/com/mraof/minestuck/computer/editmode/ServerEditHandler.java

```diff
@@ -427,8 +430,9 @@ public static void onBlockBreak(PlayerInteractEvent.LeftClickBlock event)
 			EditData data = getData(event.getPlayer());
 			if(!MinestuckConfig.SERVER.gristRefund.get())
 			{
-				GristHelper.decrease(event.getWorld(), data.connection.getClientIdentifier(), new GristSet(GristTypes.BUILD,1));
-				GristHelper.notifyEditPlayer(event.getWorld().getServer(), data.connection.getClientIdentifier(), new GristSet(GristTypes.BUILD, 1), false);
+				GristHelper.decreaseAndNotify(event.getWorld(), data.connection.getClientIdentifier(), new GristSet(GristTypes.BUILD,1), GristHelper.EnumSource.SERVER);
+				if (!data.connection.getClientIdentifier().getUsername().equals(data.connection.getServerIdentifier().getUsername()))
+					GristHelper.notify(event.getPlayer().getServer(), data.connection.getServerIdentifier(), new GristSet(GristTypes.BUILD,1), GristHelper.EnumSource.SERVER, false);
```

Comment:
Three comments that applies to all of these new notifications:
- You can compare player identifiers directly with `==` and `!=`.
- You're notifying the server player. However who you want to notify is the edit player, which is usually, but *not necessarily* the same player as the server player.
- This seems to apply to all notifications with the `SERVER` source. Do you think it would make sense to implement this extra notification into `GristHelper.notify()` or such instead?

Reply:
Moved it all into notify() and changed the player from the server to the editmode user.